src/main/java/intellistart/interviewplanning/model/booking/BookingRepository.kt

```
@@ -0,0 +1,70 @@
+package intellistart.interviewplanning.model.booking
+
+import intellistart.interviewplanning.model.slot.Slot
+import intellistart.interviewplanning.model.user.User
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.aggregation.Aggregation
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.query.Update
+import org.springframework.stereotype.Repository
+
+@Repository
+class BookingRepository(private val mongoTemplate: MongoTemplate) {
+
+    fun findById(id: ObjectId): Booking? {
```

instead of return nullable type, try throw custom exception (**BookingByIdIsMissingException** for example)

---

I do it on a service layer

---

maybe it would better to throw exception immediately on repository level, instead of work with nullability on service layer😅

---

no, service layer is more appropriate place to throw *NotFoundException